code/modules/surgery/implant.dm

```diff
@@ -152,9 +152,14 @@
 	tool.forceMove(target)
 
 	if(istype(tool, /obj/item/weapon/implant))
-		var/obj/item/weapon/implant/disobj = tool
-		disobj.part = affected
-		affected.implants += disobj
+		var/obj/item/weapon/implant/timp = tool
+		timp.part = affected
+		affected.implants += timp
+		if(timp.implanted(target, user))
+			timp.forceMove(target)
+			timp.imp_in = target
+			timp.implanted = 1
+			timp.part = affected
```

Comment:
Remember to copypaste the check done here https://github.com/vgstation-coders/vgstation13/blob/8bbdb937afd32a18c3fd86e2e36815f3800c41df/code/game/objects/items/weapons/implants/implant.dm#L274 

Verify that the implant is a loyalty one first ofc.

Reply:
oh also here the spawn(1) won't be necessary

Reply:
`if(timp.implanted(target, user))` runs that check already